### PR TITLE
Prepare for dropping Swift 5.3

### LINKED ISSUE #523: [discussion] Drop Swift 5.3 support
## Suggestion
- Make current version (`v3.12.3`) to be a last support for Swift 5.3
- Require Swift 5.4 for next release (`v3.12.4` or `v3.13.0`).
- Soon, create a new release ( `v3.12.5`, `v3.13.0` or `v3.13.1` ) and make it to require Swift 5.5.

## Motivation
There is an issue and workaround for building Danger-Swift in Swift 5.3 Docker image (https://github.com/danger/swift/issues/375, https://github.com/danger/swift/pull/426).

Fortunately, I tried to run `swift build --disable-sandbox -c release` in Swift 5.4 image and succeeded!
I think that supporting Swift 5.3 is no longer necessary.

Additionally, dropping Swift 5.4 is more happier for CI because
- Xcode 13.3 requires macOS monterey
- macOS monterey runner in GitHub Actions doesn't have Xcode 12.x (https://github.com/actions/virtual-environments/blob/main/images/macos/macos-12-Readme.md#xcode)
  - it means Swift 5.4 is not supported in macOS monterey runner.

### PR DESCRIPTION
**Pull request by 417-72KI**
Resolves #523

### CODE REVIEW

File: Scripts/install.sh [hunk 1]
```diff
@@ -1,48 +1,26 @@
...
-        swift build --disable-sandbox --arch arm64 -c release
-    else
-        swift build --disable-sandbox -c release
-    fi
-else
+if [ $? -ne 0 ]; then
+    echo '[WARN] Failed to install with `release` configuration. Try to install with `debug` configuration.'
     BUILD_FOLDER=".build/debug"
-    swift build --disable-sandbox
+    swift build --disable-sandbox -c debug
```
Comment (417-72KI) [inline Scripts/install.sh:23]: [`make install` was failed in 5.5.3](https://github.com/danger/swift/runs/6988748337?check_suite_focus=true) so workaround should be still alive 🙄
However, [`make install` in 5.4.3 succeeds](https://github.com/danger/swift/runs/6988748235?check_suite_focus=true) and it should not switch configuration by Swift version.

File: Scripts/install.sh [hunk 2]
```diff
@@ -1,48 +1,26 @@
...
-then
-    BUILD_FOLDER=".build/release"
-    if [[ `uname -m` == 'arm64' ]]; then
-        swift build --disable-sandbox --arch arm64 -c release
-    else
-        swift build --disable-sandbox -c release
-    fi
-else
+if [ $? -ne 0 ]; then
+    echo '[WARN] Failed to install with `release` configuration. Try to install with `debug` configuration.'
```
[the remark this answers is not in the document]
Comment (f-meloni) [reply Scripts/install.sh]: I'm a bit torn about this.
Danger is meant to run on CI, where even small time differences might matter, hence I would like if possible to keep it as fast as possible.
Given it fails only on Linux, what about `swift build --disable-sandbox` if we are on Linux?

[the remark this answers is not in the document]
Comment (417-72KI) [reply Scripts/install.sh]: Right. I'm going to fix it.

**Comment by f-meloni** [comment]
Amazing! Thank you :)

**Pull request merged**